addons/mrp/static/src/views/mrp_documents_kanban/mrp_documents_kanban_controller.js

```diff
@@ -4,10 +4,11 @@ import { patch } from "@web/core/utils/patch";
 import { ProductDocumentKanbanController } from "@product/js/product_document_kanban/product_document_kanban_controller";
 
 patch(ProductDocumentKanbanController.prototype, {
-    buildFormData(formData) {
-        super.buildFormData(formData);
+    buildFormData() {
+        let res = super.buildFormData();
         if (this.props.context.attached_on_bom) {
-            formData.append("attached_on_bom", this.props.context.bom_id);
+            res["attached_on_bom"] = this.props.context.bom_id;
         }
+        return res;
```

Comment:
This will break this override: `addons/sale_pdf_quote_builder/static/src/js/quotation_document_kanban/quotation_document_kanban_widget.js`

I'd rather have you patch `formData` directly so we don't have to change every other override:
```suggestion
    setup() {
        super.setup(...arguments);
        if (this.props.context.attached_on_bom) {
            this.formData.attached_on_bom = this.props.context.bom_id;
            // this. allowedMIMETypes = 'application/pdf';
        }
```
FYI you can also restrict specific MIME Type if that suits your use case by defining `allowedMIMETypes`